src/Tizen.Multimedia/AudioManager/AudioStreamPolicy.cs

```diff
@@ -304,6 +306,118 @@ public void RemoveDeviceForStreamRouting(AudioDevice device)
+        /// <since_tizen> 6 </since_tizen>
+        public AudioDevice PreferredInputDevice
+        {
+            get
+            {
+                if (_inputDevice == null)
+                {
+                    return null;
+                }
+
+                Interop.AudioStreamPolicy.GetPreferredDevice(Handle, out var inDeviceId, out _).
+                    ThrowIfError("Failed to get preferred input device");
+
+                if (inDeviceId != _inputDevice.Id)
+                {
+                    throw new InvalidOperationException("output device id does not match");
+                }
+
+                return _inputDevice;
+            }
+            set
+            {
+                int deviceId;
+
+                if (value == null)
+                {
+                    deviceId = 0;
+                }
+                else
+                {
+                    deviceId = value.Id;
+                }
+
+                Interop.AudioStreamPolicy.SetPreferredDevice(Handle, AudioDeviceIoDirection.Input, deviceId).
+                    ThrowIfError("Failed to set preferred input device");
+
+                _inputDevice = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the preferred output device.
+        /// </summary>
+        /// <value>
+        /// The <see cref="AudioDevice"/> instance.<br/>
+        /// The default is null which means any device is not set on this property.
+        /// </value>
+        /// <remarks>
+        /// This property is to set a specific built-in device when the system has multiple devices of the same built-in device type.
+        /// When there's only one device for a built-in device type in the system, nothing will happen even if this property is set successfully.
+        /// </remarks>
+        /// <exception cref="ArgumentException">A device is not for output.</exception>
+        /// <exception cref="InvalidOperationException">An internal error occurs.</exception>
+        /// <exception cref="AudioPolicyException">A device is not supported by this <see cref="AudioStreamPolicy"/> instance.</exception>
+        /// <exception cref="ObjectDisposedException">The <see cref="AudioStreamPolicy"/> has already been disposed of.</exception>
+        /// <seealso cref="AudioManager.GetConnectedDevices()"/>
+        /// <since_tizen> 6 </since_tizen>
+        public AudioDevice PreferredOutputDevice
+        {
+            get
+            {
+                if (_outputDevice == null)
+                {
+                    return null;
+                }
+
+                Interop.AudioStreamPolicy.GetPreferredDevice(Handle, out _, out var outDeviceId).
+                    ThrowIfError("Failed to get preferred output device");
+
+                if (outDeviceId != _outputDevice.Id)
+                {
+                    throw new InvalidOperationException("output device id does not match");
```

Comment:
I don't understand, Why you need throw exception? 
When does this happen?

I expect, a AudioDevice instance should be returned, and it's device id is equal to _outputDevice.Id

Reply:
As you know, setting the property is performed via native API only with device id and it affects sound daemon routing path. Don't we need to check again via native API it in case of getting the property? (e.g. If the daemon crash happens after setting the preferred device, it would have invalid value. Of course, it is abnormal case.)

Reply:
Yes, I think, no need to check from native.
If set operation was failed, exception should be occurred in setter.
but if not occurred exception, API must return a value that is set by user

Reply:
I think we have a misunderstanding. I meant the pinvoke in "getter" property could give different value in such cases I wrote above, so I added the condition to check in the getter property.

Reply:
Is it a possible?
A value is applied to instance of AudioPolice that is created by c# code
How to different value was return from native?

Reply:
It could be possible via native API. (I described it in the previous comments regarding inconsistent with system status for exception case.)

Reply:
If multimedia systems are in inconsistent state, Native call will be failed. so, `ThrowIfError("Failed to get preferred output device");`  exception should be raised, and do not reached here

Reply:
I've been preparing the patch like that including core service. It could be removed after all.